lib/modules/platform/github/index.ts

```diff
@@ -1910,25 +1911,25 @@ export async function mergePr({
     }
   }
   if (!automerged) {
-    // We need to guess the merge method and try squash -> rebase -> merge
-    options.body.merge_method = 'rebase';
+    // We need to guess the merge method and try squash -> merge -> rebase
```

Comment:
@rarkins we should use this breaking change to only try the defined/ configured merge method and don't retry other methods. we don't do it on other platforms.

Reply:
Are we able to reliably determine it?

Reply:
As far as I understand, yes. This logic is already done during the `initRepo` when we check `if (repo.squashMergeAllowed)` etc etc. The query uses the GraphQL API which has the information on allowed merging methods on the repo (I just checked the [query](https://github.com/renovatebot/renovate/blob/6c1bd70bfd4c528f8fc50b5d24c19a69b4b9a0ac/lib/modules/platform/github/graphql.ts#L2) and the [doc](https://docs.github.com/en/graphql/reference/objects#repository))

Reply:
BTW this code path which is changed is only executed if config.merge_method is unknown or the merge failed. But then you mention here that it can be reliably determined

Reply:
Yeah, that's on me. You will reliably determine a merge method during the `initRepo`, but only if the token _has the right to know_ (there is a comment mentioning non-admins won't be able to read the merge methods allowed for a repo). Which means, indeed, we can't reliably know.

If this block of code is to be removed, then it would mean Renovate would have to require the set of permissions (which I do not explicitly know sadly) required to know the merge methods allowed. From my experience testing yesterday, the official app already has the rights to this information, but then self-hosted is another thing.

Reply:
Gotcha. In this case we need to keep trying multiple methods for the same reason